spec/unit/changed_managed_files_inspector_spec.rb

```diff
@@ -18,42 +18,97 @@
-  let(:stat_result) { File.read("spec/data/changed_managed_files/stat_result") }
-  let(:system) { double }
-  let(:description) {
-    SystemDescription.new("foo", SystemDescriptionStore.new)
+  let(:rpm_database) { double }
+
+  let(:system) {
+    double(
+      rpm_database:                      rpm_database,
+      check_requirement:                 true,
+      check_retrieve_files_dependencies: true
+    )
   }
+  let(:description) { SystemDescription.new("foo", SystemDescriptionStore.new) }
   let(:filter) { nil }
-  subject {
-    inspector = ChangedManagedFilesInspector.new(system, description)
-
-    allow(system).to receive(:check_requirement).at_least(:once)
-    allow(system).to receive(:run_script) do |*script, options|
-      expect(script.first).to eq("changed_files.sh")
-      options[:stdout].puts rpm_result
-    end
 
-    allow(system).to receive(:run_command).with("stat", "--printf",
-      "%a:%U:%G:%u:%g:%F:%n\\n", "/etc/iscsi/iscsid.conf",
-      "/etc/apache2/de:fault server.conf", "/etc/apache2/listen.conf",
-      "/usr/share/man/man1/time.1.gz", "/usr/bin/cron'tab",
-      anything).and_return(stat_result)
-    allow(system).to receive(:run_command).with("find", "/usr/bin/cron'tab", any_args).
-      and_return("/etc/fo'o")
-
-    inspector
-  }
+  subject { ChangedManagedFilesInspector.new(system, description) }
 
   describe "#inspect" do
-    silence_machinery_output
+    before(:each) do
+      allow(rpm_database).to receive(:changed_files).and_return(
+        [
+          RpmDatabase::ChangedFile.new("c",
+            name:            "/etc/config",
+            status:          "changed",
+            changes:         ["md5"],
+            package_name:    "zypper",
+            package_version: "1.6.311"
+          ),
+          RpmDatabase::ChangedFile.new("",
+            name:            "/etc/file",
+            status:          "changed",
+            changes:         ["deleted"],
+            package_name:    "zypper",
+            package_version: "1.6.311"
+          ),
+          RpmDatabase::ChangedFile.new("",
+            name:            "/etc/dir",
+            status:          "changed",
+            changes:         ["link_path"],
+            package_name:    "zypper",
+            package_version: "1.6.311"
+          ),
+          RpmDatabase::ChangedFile.new("",
+            name:            "/etc/documentation",
+            status:          "changed",
+            changes:         ["user"],
+            package_name:    "zypper",
+            package_version: "1.6.311"
+          ),
+          RpmDatabase::ChangedFile.new("",
+            name:            "/usr/share/man/man1/time.1.gz",
+            status:          "changed",
+            changes:         ["user"],
+            package_name:    "man",
+            package_version: "2"
+          ),
+        ]
+      )
+      allow(rpm_database).to receive(:get_path_data).and_return(
+          "/etc/file" =>                     {
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Indent the first parameter one step more than the start of the previous line.